### DISCUSSION TITLE
Getting the error "CorsairLink could not initialize or has no sensors" on H100i V2

### DISCUSSION THREAD

**Discussion by stayve12** [EvanMulawski/FanControl.CorsairLink#15] [General]
I didn't see the H100i V2 in the list of supported water coolers.  Any Idea if mine is supported? I get the "CorsairLink could not initialize or has no sensors" error when opening FanControl.  All I did was copy FanControl.CorsairLink.dll to FanControl's Plugins folder like the readme said. Is that all that was needed? I am not supposed to do anything with the folders:  FanControl.CorsairLink-main > CorsairLink.Abstractions and all of the .cs files that are inside?

Any help would be greatly appreciated.  I made sure that CorsairService was not running.  My BIOS is set to PWM for the CPU pump and CPU fan, then DC for my case fans which FanControl can control fine. BIOS has smart fan turned off too and just a static rpm set.  

Side note - I've completely had it with Corsair iCue.  The newest version 4.x.x and legacy 3.x.x both control my CPU pump and fans fine, but the app will freeze all RPM levels and stop controlling my CPU fans until I restart the service in iCue settings.  

I can't control my CPU pump or fans with the BIOS, MSI Afterburner, or using this plugin for FanControl and Corsair's proprietary iCue software is unreliable at best.  I'm about to just buy a new water cooler at this point :/

**Comment by EvanMulawski** [comment]
Hi @stayve12 - can you post your log.txt file (located in the Fan Control directory)?

And yes, only the dll file from the release ZIP file is needed.

**Reply by stayve12** [reply]
Sure heres the log file.  Also I have been trying FanControl out while iCue 3.x.x is currently installed but not running.  I can uninstall iCue if you think that might make a difference.
[log.txt](https://github.com/EvanMulawski/FanControl.CorsairLink/files/10716803/log.txt)

**Reply by EvanMulawski** [reply]
Interesting - it's not showing up in the log as a Corsair device. Can you download and run [USBDeview](https://www.nirsoft.net/utils/usbdeview-x64.zip) to try to find it? The Vendor ID should be `1b1c`. If you do find it, copy the row and paste it here.

**Reply by stayve12** [reply]
From USB Deview Vendor ID matching 1b1c:

H100i v2	Corsair HydroSeries H100i v2	Vendor Specific	Yes	Yes	No	No		7289_2.0	2/11/2023 11:14:41 PM	1/18/2023 6:35:18 PM	1b1c	0c09	1.00		ff	00	00		BATTLESTATION				SIUSBXP		SiUSBXp.sys		Corsair Components, Inc.	@oem46.inf,%FRIENDLYNAME%;Corsair USBXp Driver	50 mA	1.10	Corsair HydroSeries H100i v2	3.3.0.0	DriverInstall.NT	oem46.inf	USB\VID_1B1C&PID_0C09\7289_2.0	Removable, UniqueID, SurpriseRemovalOK

**Reply by EvanMulawski** [reply]
So `0c09` is under the Asetek driver family, which isn't supported. These devices are UART and not HID, which is why the plugin did not list your device - it's a different protocol that requires a different communication library (Silicon Labs USBXpress SDK). I'm not sure it's worth the time to add support for this as those devices are discontinued and testing would be impossible.

**Reply by stayve12** [reply]
Dang that sucks.  Thank you for the assistance though!  Guess i'll just swap to an air cooler at this point haha

**Reply by EvanMulawski** [reply]
@stayve12 I was able to add support for the Asetek Hydro AIOs if you still have yours. Support is available in [v1.4.0-beta.1](https://github.com/EvanMulawski/FanControl.CorsairLink/releases/tag/v1.4.0-beta.1).

**Reply by stayve12** [reply]
Hey sorry it's been so long, just saw this now. I went back to using Corsair Link 4 to control my H100i V2 consistently since Corsair iCue would freeze up for me randomly before. I have been using FanControl to control my case and GPU fans, can confirm V177 still does not work to control CPU cooler & pump fans.

I'm willing to give this a shot again if there is a chance FanControl can control everything and I can stop using a 10+ yr old piece of software (Link4) haha. Should I download the v1.4-beta that you linked or is there a newer version?

**Reply by EvanMulawski** [reply]
Hi @stayve12, the latest release should work for you. See [Releases](https://github.com/EvanMulawski/FanControl.CorsairLink/releases).

**Reply by stayve12** [reply]
It works perfectly, you absolute legend thank you!!!
Is there any specific update process I should follow for this or just let FanControl continue to auto-update?

**Reply by EvanMulawski** [reply]
Check that Releases page occasionally - Fan Control plugins do not auto-update.

**Reply by stayve12** [reply]
Actually one thing I noticed that does not work anymore. When I start FanControl, it turns off the RGB backlit Corsair logo on my cooler. Just booting up the PC with no Corsair Link or iCue program running it will light up red (color I chose with Link a while ago) but once I start FanControl it turns off the RGB. Any way to change that?

**Reply by EvanMulawski** [reply]
Try OpenRGB.

**Comment by xTSwizzle** [comment]
Hey, I have a Corsair iCUE H115i Elite Capellix and an Asus Z590 Gaming E Motherboard. 

I had my FanControl all setup perfectly. Today I was messing around in my BIOS and when I rebooted I got the same error message posted here. I had changed my Asus Aura setting in my BIOS from "ALL ON" to "STEALTH ONLY". Switching back to "ALL ON" allowed this Plugin to properly read the sensors again! Unsure if this is helpful, but thought I'd chime in. 

Log attached just in case. 

[log.txt](https://github.com/EvanMulawski/FanControl.CorsairLink/files/10787255/log.txt)

**Reply by EvanMulawski** [reply]
Moved to #17
